services/galley/src/Galley/API/Action.hs

```diff
@@ -749,39 +758,78 @@ notifyConversationAction tag quid notifyOrigDomain con lconv targets action = do
         Set.difference
           (Set.map void (bmRemotes targets))
           (Set.fromList (map (void . rmId) (convRemoteMembers conv)))
-  subConvs <- Map.assocs <$> E.listSubConversations (convId conv)
-  E.runFederatedConcurrently_ (toList newDomains) $ \_ -> do
-    void $
-      fedClient @'Galley @"on-new-remote-conversation"
+  let nrc =
         NewRemoteConversation
           { nrcConvId = convId conv,
             nrcProtocol = convProtocol conv
           }
-    for_ subConvs $ \(mSubId, mlsData) ->
-      fedClient @'Galley @"on-new-remote-subconversation"
-        NewRemoteSubConversation
-          { nrscConvId = convId conv,
-            nrscSubConvId = mSubId,
-            nrscMlsData = mlsData
-          }
+  subConvs <- Map.assocs <$> E.listSubConversations (convId conv)
 
-  update <- fmap (fromMaybe (mkUpdate []) . asum . map tUnqualified)
-    . E.runFederatedConcurrently (toList (bmRemotes targets))
-    $ \ruids -> do
-      let update = mkUpdate (tUnqualified ruids)
-      -- if notifyOrigDomain is false, filter out user from quid's domain,
-      -- because quid's backend will update local state and notify its users
-      -- itself using the ConversationUpdate returned by this function
-      if notifyOrigDomain || tDomain ruids /= qDomain quid
-        then fedClient @'Galley @"on-conversation-updated" update $> Nothing
-        else pure (Just update)
+  let errorIntolerant = do
+        E.runFederatedConcurrently_ (toList newDomains) $ \_ -> do
+          void $ fedClient @'Galley @"on-new-remote-conversation" nrc
+          for_ subConvs $ \(mSubId, mlsData) ->
```

Comment:
We had lots of merge conflicts because code was changed around places where on MLS we need to be aware and account for subconversations. During merge, the sub conversation notification flow for this function got swallowed up, and took me a bit to notice. Luckily this was immediately caught by an integration test, so at least no lasting bug was introduced.